docs/kb/semgrep-supply-chain/ssc-lockfiles-circleci.md

```diff
@@ -0,0 +1,62 @@
+---
+description: How to generate lockfiles for Semgrep Supply Chain in a Circle CI pipeline.
+tags:
+  - Semgrep Supply Chain
+  - CircleCI
+  - Lockfiles
+  - Workspaces
+---
+## Generate lockfiles for Semgrep Supply Chain in a Circle CI pipeline
+
+Semgrep Supply Chain needs your project's lockfiles as input to scan your codebase successfully. If the [lockfiles that Supply Chain supports](/docs/supported-languages/#semgrep-supply-chain) are not under source control in your project, you can generate the lockfile as part of the CI job.
+
+In CircleCI, you can generate a lockfile during the first job and then pass it to the Semgrep scan using a [workspace](https://circleci.com/docs/workspaces/) to share files between jobs.
+
+The following `config.yml` file demonstrates how you can generate a lockfile and pass it to subsequent jobs using CircleCI workspaces. This example uses a `maven_dep_tree.txt` file, which [typically needs to be generated](/docs/semgrep-supply-chain/setup-maven/) from a `pom.xml` for Maven dependency tracking.
+
+```yaml
+version: 2.1
+
+jobs:
+  build:
```

Comment:
Maybe we should give this a more specific name than `build`?

Reply:
yes, I agree. I changed that to lock_file_generation. Thanks @armchairlinguist and @khorne3!!